app/models/batch_connect/session.rb

```diff
@@ -182,14 +205,22 @@ def script(opts = {})
         # hopefully lsf also sets the shell from the shebang
       end
 
-      OodCore::Job::Script.new(
-        {
-          content: script_content(bc_opts),
-          job_name: job_name,
-          workdir: output_root,
-          output_path: output_file
-        }.merge script_opts
+      script_opts = {
```

Comment:
> Its only state is the `cluster.job_config[:adapter]`, which arguably should be an argument as well.

It also has `job_name`, `staged_root`, and `output_file` as state.

> Probably more appropriate would be a separate method that accepts `OodCore::Job::Script` instance as an argument and writes the data to the files.

The `OodCore::Job::Script` isn't serializable by design. So this wouldn't work, unless we refactored it to be serializable.

Another solution would be to create a new method called `Session#script_opts` that generates the initialization hash that is then fed to `OodCore::Job::Script#initialize`.